models/project_board.go

```diff
@@ -0,0 +1,223 @@
+
+	case None:
+		fallthrough
+	default:
+		return nil
+	}
+
+	if len(items) == 0 {
+		return nil
+	}
+
+	var boards = make([]ProjectBoard, 0, len(items))
+
+	for _, v := range items {
+		boards = append(boards, ProjectBoard{
+			CreatedUnix: timeutil.TimeStampNow(),
+			UpdatedUnix: timeutil.TimeStampNow(),
+			CreatorID:   project.CreatorID,
+			Title:       v,
+			ProjectID:   project.ID,
+		})
+	}
+
+	_, err := sess.Insert(boards)
+	return err
+}
+
+// NewProjectBoard adds a new project board to a given project
+func NewProjectBoard(board *ProjectBoard) error {
+	_, err := x.Insert(board)
+	return err
+}
+
+// DeleteProjectBoardByID removes all issues references to the project board.
+func DeleteProjectBoardByID(boardID int64) error {
+	sess := x.NewSession()
+	defer sess.Close()
+	if err := sess.Begin(); err != nil {
+		return err
+	}
+
+	if err := deleteProjectBoardByID(sess, boardID); err != nil {
+		return err
+	}
+
+	return sess.Commit()
+}
+
+func deleteProjectBoardByID(e Engine, boardID int64) error {
+	board, err := getProjectBoard(e, boardID)
+	if err != nil {
+		if IsErrProjectBoardNotExist(err) {
+			return nil
+		}
+
+		return err
+	}
+
+	if err = board.removeIssues(e); err != nil {
+		return err
+	}
+
+	if _, err := e.ID(board.ID).Delete(board); err != nil {
+		return err
+	}
+	return nil
+}
+
+func deleteProjectBoardByProjectID(e Engine, projectID int64) error {
+	_, err := e.Where("project_id=?", projectID).Delete(&ProjectIssues{})
+	return err
+}
+
+// GetProjectBoard fetches the current board of a project
+func GetProjectBoard(boardID int64) (*ProjectBoard, error) {
+	return getProjectBoard(x, boardID)
+}
+
+func getProjectBoard(e Engine, boardID int64) (*ProjectBoard, error) {
+	board := new(ProjectBoard)
+
+	has, err := e.ID(boardID).Get(board)
+	if err != nil {
+		return nil, err
+	} else if !has {
+		return nil, ErrProjectBoardNotExist{BoardID: boardID}
+	}
+
+	return board, nil
+}
+
+// UpdateProjectBoard updates the title of a project board
+func UpdateProjectBoard(board *ProjectBoard) error {
+	return updateProjectBoard(x, board)
+}
+
+func updateProjectBoard(e Engine, board *ProjectBoard) error {
+	board.UpdatedUnix = timeutil.TimeStampNow()
+	_, err := e.ID(board.ID).Cols(
+		"title",
+		"default",
+		"updated_unix",
+	).Update(board)
+	return err
+}
+
+// GetProjectBoards fetches all boards related to a project
+func GetProjectBoards(projectID int64) ([]*ProjectBoard, error) {
+
+	var boards = make([]*ProjectBoard, 0, 5)
+
+	sess := x.Where("project_id=?", projectID)
+	return boards, sess.Find(&boards)
+}
+
+// GetUnCategorizedBoard represents a board for issues not assigned to one
+func GetUnCategorizedBoard(projectID int64) (*ProjectBoard, error) {
+	return &ProjectBoard{
+		ProjectID: projectID,
+		Title:     "UnCategorized",
+		Default:   true,
+	}, nil
+}
+
+// LoadIssues load issues assigned to this board
+func (b ProjectBoard) LoadIssues() (IssueList, error) {
```

Comment:
How big can a board get? Are we genuinely only looking at say 10 issues or is it possible that this could be 100s of issues?

 Should we not add some paging?

Reply:
github has no pagination on project borards - could not see one

if we do pagination it will require some smal UI changes - i think this is an enhancement witch can be done later in a seperate pull

@zeripath if you think this is an important enhancement - create a issue to have a track of it so people can work on this after this pull has merged?

Reply:
I think it's OK now. But a better resolution is a pagination on backend with a scroll loading on frontend.

Reply:
`b ProjectBoard` should be `b *ProjectBoard`, otherwise `b.Issues = issues` will be ignored.

Reply:
```suggestion
func (b *ProjectBoard) LoadIssues() (IssueList, error) {
```